src/core/lib/channel/metrics.h

```diff
@@ -151,6 +362,26 @@ class CallbackMetricReporter {
 
 class RegisteredMetricCallback;
 
+class CallbackMetricReporterWrapper {
```

Comment:
I think if my suggestion above pans out, then we won't need this wrapper anymore.

Reply:
Indeed we don't need this wrapper. Woohoo!